src/core/meshcop/commissioner.cpp

```diff
@@ -164,27 +164,40 @@ otError Commissioner::Start(otCommissionerStateCallback  aStateCallback,
     SetState(OT_COMMISSIONER_STATE_PETITION);
 
 exit:
+    if (error != OT_ERROR_NONE)
+    {
+        Get<Coap::CoapSecure>().Stop();
```

Comment:
@bukepo Would calling `Get<Coap::CoapSecure>().Stop()` cause any issue if `Get<Coap::CoapSecure>().Start` is not executed. This could happen if `Commissioner::Start` failed in the first two lines of `VerifyOrExit`

Reply:
If it would, it should be fixed. I think we Stop() should be a idempotent operation. Thoughts?

Reply:
Agree that `CoapSecure::Stop` should be idempotent. Couldn't find anything non-idempotent along the call tree.